notificationworkerlambda/src/main/scala/com/gu/notifications/worker/SenderRequestHandler.scala

```diff
@@ -83,28 +80,16 @@ trait SenderRequestHandler[C <: DeliveryClient] extends Logging {
       }.parJoin(maxConcurrency)
   }
 
-  def logStartAndCount(acc: Int, chunkedTokens: ChunkedTokens) = {
-    logger.info(Map(
-          "notificationId" -> chunkedTokens.notification.id
-        ), "Start processing a SQS message");
-    acc + chunkedTokens.tokens.size
-  }
-
   def handleChunkTokens(event: SQSEvent, context: Context): Unit = {
     val sqsMessageBatchSize = event.getRecords.size
     val startTime = Instant.now
-    val totalTokensProcessed: Int = event.getRecords.asScala.map(event => NotificationParser.parseChunkedTokenEvent(event.getBody)).foldLeft(0)(logStartAndCount)
     val chunkedTokenStream: Stream[IO, (ChunkedTokens, Long, Instant, Int)] = Stream.emits(event.getRecords.asScala)
       .map(r => (r.getBody, r.getAttributes.getOrDefault("SentTimestamp", "0").toLong))
       .map { case (body, sentTimestamp) => (NotificationParser.parseChunkedTokenEvent(body), sentTimestamp, startTime, sqsMessageBatchSize) }
 
-    logger.info("Java version: " + System.getProperty("java.version"))
-
     deliverChunkedTokens(chunkedTokenStream)
       .compile
       .drain
       .unsafeRunSync()
-
-    logEndOfInvocation(sqsMessageBatchSize, totalTokensProcessed, startTime, registry)
```

Comment:
Do you think if it would be useful to have a start and end log with notification ID for each lambda invocation?  I think they may help us to understand when each lambda invocation starts and how long it takes.  They may give us some finer level of information on the concurrency of lambda execution.

Reply:
Good point, maybe I've been too rash with what I've removed...
- in terms of lambda invocation times we get that info from the standard aws metrics
- however, given that I remove this, we would lose the trace in kibana

Looking at Jacob's PR I think we might have a solution for understanding overall latency metrics, but maybe it is good to understand latency on a per-notificationId basis too via kibana (for extra debugging)

Reply:
I've added the start and end log messages back in, I think the log trace looks sensible (for now):

Dry run:
<img width="1497" alt="Screenshot 2023-01-03 at 12 01 57" src="https://user-images.githubusercontent.com/45561419/210354456-54acbcd5-7460-4460-a5f3-23cca85abb36.png">
^ we can see the beginning and end of the processing by notification id

Not a dry run (sent to only a few subscribers):
<img width="1496" alt="Screenshot 2023-01-03 at 12 03 47" src="https://user-images.githubusercontent.com/45561419/210354512-69d28cb6-c29a-422d-9c39-850392df1ee0.png">
^ we continue to log for every api call, but I plan to address this after #884 is complete as it will change what metrics we're capturing and how.

Without filtering by notification id we also see when all sqs messages in a single event have been processed too:
<img width="1496" alt="Screenshot 2023-01-03 at 12 09 30" src="https://user-images.githubusercontent.com/45561419/210354862-d18d50ed-c215-4e2e-9277-ea9c2dd9b372.png">